### ISSUE TITLE
Rendering error when frame bounds don't fit in logical screen bounds

### ISSUE THREAD

**Issue by thomaslevesque** [XamlAnimatedGif/XamlAnimatedGif#71]
As described by @x000000 in #70 

For instance, in [this image](https://cdn.betterttv.net/emote/55ae376ed5e0264366129a7d/1x), the logical screen has a size of `(28, 28)`, but the second frame has its top-left corner in `(2, 3)` and a size of `(26, 28)`, which doesn't fit in the logical screen. This causes the call to `AddDirtyRect` to fail:

```
System.ArgumentException: Value does not fall within the expected range.
   at MS.Internal.MILSwDoubleBufferedBitmap.AddDirtyRect(SafeMILHandle THIS_PTR, Int32Rect& dirtyRect)
   at System.Windows.Media.Imaging.WriteableBitmap.AddDirtyRect(Int32Rect dirtyRect)
   at XamlAnimatedGif.Animator.<RenderFrameAsync>d__46.MoveNext() in D:\Docs\GitHub\XamlAnimatedGif\XamlAnimatedGif.Shared\Animator.cs:line 350
```

**Comment by x000000** [comment]
I've tried to redraw every frame completely (i.e. set AddDirtyRect(new Int32Rect(0, 0, 28, 28))), but there are some artifacts on few frames that should be fixed somehow. So setting valid dirty rect is only a part of an issue.

**Comment by thomaslevesque** [comment]
@x000000, you're right, it's not enough to fix the dirty rect size. Other usages of frame width/height in `RenderFrameAsync` probably need to be fixed as well, otherwise it could end up overwriting other parts of the image. Working on it now.

**Comment by thomaslevesque** [comment]
OK, it's a bit more subtle than I thought...

The image declares a frame rect of `(2,3,26,28)`, which is incorrect... but the number of pixels in the image data is actually 728 (26 * 28), which matches the information in the incorrect frame descriptor. So we have to read the pixels from the image data as specified in the descriptor, but write to the WritableBitmap buffer according to the "corrected" frame size.

I got it working, will push in a few minutes unless I notice another issue.

**Comment by thomaslevesque** [comment]
This should be fixed in 1.1.3. The new NuGet package will be online shortly.
